logdna/adapter/adapter.go

```diff
@@ -153,47 +141,45 @@ func (adapter *Adapter) Stream(logstream chan *router.Message) {
 // readQueue is a method for reading from queue:
 func (adapter *Adapter) readQueue() {
 
-    buffer := make([]Line, 0)
-    timeout := time.NewTimer(adapter.Limits.FlushInterval)
-    byteSize := 0
+    buffer := make([]Line, 0, adapter.Config.MaxBufferSize)
```

Comment:
You are allocating a pretty big buffer here `sizeOf(Line) *  adapter.Config.MaxBufferSize`.

You may want to start with an empty buffer and let `append()` grow it dynamically.

Reply:
Updated!